autogen/agentchat/conversable_agent.py

```diff
@@ -1283,7 +1451,10 @@ def generate_init_message(self, **context) -> Union[str, Dict]:
         Args:
             **context: any context information, and "message" parameter needs to be provided.
         """
-        return context["message"]
+        message = context["message"]
+        if isinstance(message, str) or isinstance(message, list):
+            message = {"content": message}
+        return message
```

Comment:
@sonichi @yenif this change is causing the notebook test to fail because the in notebook https://github.com/microsoft/autogen/blob/main/notebook/agentchat_auto_feedback_from_code_execution.ipynb

```python
def generate_init_message(self, *args, **kwargs) -> Union[str, Dict]:
        return super().generate_init_message(*args, **kwargs) + """
If you suggest code, the code will be executed in IPython."""
```

My feeling is that having different return types for a function is always problematic. Consider removing the string return type and just have dictionary return type for now. I think @davorrunje and others are looking into enforcing the types using Pydantic and Mypy.

Reply:
> having different return types for a function is always problematic.

absolutely

Reply:
Makes since, thanks for checking it!

I'll update the type signature and adjust the notebook

Reply:
This change will affect many other files. You can find them by searching "generate_init_message" in the repo. Would you like to consider an alternative way which doesn't require modifying other files?

Reply:
yeah not required, can just undo this chunk